Source/dx.cpp

```diff
@@ -200,14 +201,14 @@ void dx_cleanup()
 	sgdwLockCount = 0;
 	MemCrit.unlock();
 
-	if (pal_surface == nullptr)
+	if (PalSurface == nullptr)
 		return;

```

Comment:
Is it ever the case that `PalSurface == nullptr`?

Reply:
Yes if `SDL_GetWindowSurface` or `SDL_CreateRGBSurfaceWithFormat` fails, since we do not appear to check this.

Reply:
It doesn't seem like a good idea to cut `dx_cleanup` short for this reason.

I added error checking for `SDL_CreateRGBSurfaceWithFormat`; I'll do the same for `SDL_GetWindowSurface`.